src/libostree/ostree-sysroot-deploy.c

```diff
@@ -3614,25 +3614,58 @@ ostree_sysroot_deployment_set_kargs_in_place (OstreeSysroot     *self,
   if (!_ostree_sysroot_ensure_writable (self, error))
     return FALSE;
 
-  g_assert (!ostree_deployment_is_staged (deployment));
+  // handle staged deployment
+  if (ostree_deployment_is_staged (deployment))
+    {
+      /* Read the staged state from disk */
+      glnx_autofd int fd = -1;
+      if (!ot_openat_ignore_enoent (AT_FDCWD, _OSTREE_SYSROOT_RUNSTATE_STAGED, &fd, error))
+        return FALSE;
 
-  OstreeBootconfigParser *new_bootconfig = ostree_deployment_get_bootconfig (deployment);
-  ostree_bootconfig_parser_set (new_bootconfig, "options", kargs_str);
+      if (fd == -1)
+        return FALSE;
```

Comment:
I think this case can't (shouldn't) happen, because we can only have a staged deployment if the file exists.

So I'd change this to use `glnx_openat` which does *not* ignore not found errors.